### PR TITLE
Feat 81 fast rollbacks

### PR DESCRIPTION
**Pull request by chesleybrown**
Ref #81 

# Deployment/Rollback Workflow

1. If `strategy == fast-rollback` then continue
1. Rename deployment to `<name>-<deployId>` to prevent updating the current live deployment
1. Suppress deploying any `service` manifests until after all deployment manifests are available
1. If `<name>-<deployId>` is not already deployed, then Deploy as usual, otherwise skip deployment
1. If deploy fails then skip (fail as usual), otherwise continue
1. Verify deployment is fully available
1. Verify `service` selectors does select pods running in the cluster
1. Deploy all service manifests
1. If NOT rollback, delete n-3 deployments that are older than `deployId=$deployId` deployment `creationTimestamp`
1. If rollback, delete all deployments newer than deploy=$deployId deployment `creationTimestamp`

# Testing

- Added functional tests to verify deployments using the fast-rollback strategy work by keeping the deployments in reserve and deleting them as the number in reserve increases
- Added functional tests to verify rolling back to a deployment available in the reserve works as expected and deletes the newer deployments based on creation time

# TODO

- [x] Add unit test to verify rolling back to a deployment that isn't available in reserve will still rollback correctly

### CODE REVIEW

**Comment by brianhall** [comment]
Workflow looks good to me so far, @chesleybrown!

**Comment by chesleybrown** [comment]
I've pushed my latest changes. I need to add additional unit test coverage and more functional tests to know for sure it's all working though. Will continue tomorrow morning. Let me know if you see any issues in the meantime.

File: README.md [hunk 1]
```diff
@@ -185,6 +185,8 @@ The following environment variables are used by this service.
 | `MANIFESTS_DIR` | Set the directory where all the manifest files are. They should be grouped in folders matching the metadata.name of the cluster that you want them deployed to | yes | `/manifests` |
 | `DRY_RUN` | Will only show the diff and will not push anything to the cluster | yes | `true` |
 | `UUID` | A UUID to be used in webhooks and API requests | false | *empty* |
+| `DEPLOY_ID` | A unique id to be used in deployments | false | *empty* |
+| `STRATEGY` | The deployment strategy to use. Currently supports rolling-update, fast-rollback. | false | `rolling-update` |
```
Comment (c1freitas) [inline README.md]: Just an fyi, kubernetes calls it `RollingUpdate` CamelCase....

Comment (chesleybrown) [inline README.md]: I mean really `fast-rollback` still uses `RollingUpdate` strategy (or whatever strategy that is defined on the manifest) when deploying.

Basically these strategies are specific to `kit-deployer` and any resemblance to existing kubernetes deployment strategies are a _coincidence_. 😅

**Comment by chesleybrown** [comment]
@InVisionApp/engineering-velocity I've updated the description and there is one last test I want to add but I think this good for review now.

**Pull request merged**